qiskit_ibm_provider/transpiler/passes/scheduling/dynamical_decoupling.py

```diff
@@ -290,89 +337,117 @@ def _pad(
             )
             return
 
-        slack = time_interval - np.sum(self._dd_sequence_lengths[qubit])
-        sequence_gphase = self._sequence_phase
+        for sequence_idx, _ in enumerate(self._dd_sequences):
```

Comment:
We don't need to sort index by duration, i.e. longer one comes first?

Reply:
Its a priority sort added by user as it also depends on the min length ratios they provide.